fabric-message-api-v1/src/client/java/net/fabricmc/fabric/mixin/client/message/ClientPlayNetworkHandlerMixin.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.message;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.client.network.ClientPlayNetworkHandler;
+
+import net.fabricmc.fabric.api.client.message.v1.ClientSendMessageEvents;
+
+/**
+ * Mixin to {@link ClientPlayNetworkHandler} to listen for sending messages and commands.
+ * Priority set to 800 to inject before {@code fabric-command-api} so that this api will be called first.
```

Comment:
Did you test this specifically? Might need a higher priority cause we want our `HEAD` to be applied last wrt. the other `HEAD` injects... needs to be confirmed by testing.

Reply:
I'm not sure what you mean. What is wrong with having this mixin applied first?

Reply:
If my understanding is correct:

```
after first mixin:

<-- method start
<-- HEAD inject at priority 800
<-- method body

after second mixin

<-- method start
<-- HEAD inject at priority 1000
<-- rest of the method (i.e. previous inject and then body)
```

So to be sure I recommend checking the generated classes using `-Dmixin.debug.export=true`.

Reply:
The HEAD inject with priority 800 (allowSendCommandMessage) is executed first.
<img width="1233" alt="Screenshot 2022-12-25 at 11 57 09" src="https://user-images.githubusercontent.com/92656833/209480529-2a4579e4-cb78-449a-96d3-a43afc57aed3.png">

Reply:
All good then.